pkg/networking/cilium/templater.go

```diff
@@ -237,6 +237,11 @@ func templateValues(spec *cluster.Spec, versionsBundle *cluster.VersionsBundle)
 		val["egressMasqueradeInterfaces"] = spec.Cluster.Spec.ClusterNetwork.CNIConfig.Cilium.EgressMasqueradeInterfaces
 	}
 
+	if spec.Cluster.Spec.ClusterNetwork.CNIConfig.Cilium.RoutingMode == anywherev1.CiliumRoutingModeDirect {
+		val["tunnel"] = "disabled"
+		val["routingMode"] = "native"
```

Comment:
@g-gaston I'm wondering if we want to expose `autoDirectNodeRoutes` also which, when all nodes are connected on the same L2, means route propagation is handled by Cilium. More info at https://docs.cilium.io/en/v1.12/concepts/networking/routing/#native-routing.

Reply:
Yeah, as long as we have a real use case for it and we understand its implications (just so we can supports folks using it in the field), we can expose it
Personally I would prefer to roll these slowly so we can have time to learn them better, but if someone needs this right now, as I said, we can for it
Let's sync offline

Reply:
Given the time this has been open I think we can add the autoDirectNodeRoutes` later.